tests/ConverterTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace DI\Bridge\Silex\Test;
+
+use DI\ContainerBuilder;
+
+use stdClass;
+use DI\Bridge\Silex\Application;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class ConverterTest extends BaseTestCase
+{
+    /**
+     * @test
+     */
+    public function should_allow_arbitrary_injection_in_converter()
+    {
+        $builder = new ContainerBuilder;
+        $builder->addDefinitions([
+            'stdClass' => function () {
+                $service = new stdClass;
+                $service->foo = 'bar';
+                return $service;
+            },
+        ]);
+
+        $application = $this->createApplication($builder);
+
+        $request = Request::create('/john?some=param');
+
+        $test = $this;
```

Comment:
ah, thanks. I do a lot of JS development so I wasn't even sure about how PHP resolves $this in closures.